contrib/gitlab/config/300-gitlabsource.yaml

```diff
@@ -0,0 +1,115 @@
+# Copyright 2019 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  labels:
+    eventing.knative.dev/source: "true"
+    knative.dev/crd-install: "true"
+  name: gitlabsources.sources.eventing.knative.dev
+spec:
+  group: sources.eventing.knative.dev
+  names:
+    categories:
+    - all
+    - knative
+    - eventing
+    - sources
+    kind: GitLabSource
+    plural: gitlabsources
+  scope: Namespaced
+  subresources:
+    status: {}
+  validation:
+    openAPIV3Schema:
+      properties:
+        apiVersion:
+          type: string
+        kind:
+          type: string
+        metadata:
+          type: object
+        spec:
+          properties:
+            accessToken:
+              properties:
+                secretKeyRef:
+                  type: object
+              type: object
+            eventTypes:
+              items:
+                enum:
+                - push_events
+                - push_events_branch_filter
+                - issues_events
+                - confidential_issues_events
+                - merge_requests_events
+                - tag_push_events
+                - note_events
+                - job_events
+                - pipeline_events
+                - wiki_page_events
+                type: string
```

Comment:
I wonder if this actually makes sense to prefix these events with a `webhook:` prefix, or something like this. We are currently only using webhooks, but I can see us adding more Serverless events. I don't have a strong opinion about doing that right now, something to think about in terms of maintaining backwards compatibility easier.

Reply:
Make sense if we try to support more event source